UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Helpers/NMHelper.cs

```diff
@@ -125,5 +132,24 @@ public static async Task<byte[]> GetFileBytesFromStream(Stream stream)
             stream.Close();
             return fileBytes;
         }
+
+        public static string GetYearAndTypeFromFilenName(string fileName)
+        {
+            if (!string.IsNullOrEmpty(fileName))
+            {
+                return string.Concat(fileName.AsSpan(fileName.Length - 4), fileName.AsSpan(fileName.Length - 7, 1));
+            }
+            else
+            {
+                return string.Empty;
+            }
+
+        }
+
+        public static List<ShowFilesResponseModel> ListFilesResponseCumulative(List<BatchDetails> batchDetails)
+        {
+            return GetShowFilesResponseModel(batchDetails).OrderByDescending(x => Convert.ToDateTime(x.Attributes.FirstOrDefault(y => y.Key == "BatchPublishedDate").Value)).GroupBy(x => x.Attributes.FirstOrDefault(y => y.Key == "Data Date").Value)
```

Comment:
Required files list can be returned from here using below code:

```suggestion
            return GetShowFilesResponseModel(batchDetails)
                .GroupBy(x => x.Attributes.FirstOrDefault(y => y.Key == "Data Date").Value)
                .Select(grp => grp.First())
                .OrderByDescending(y => y.Attributes.FirstOrDefault(y => y.Key == "Year").Value)
                .OrderByDescending(x => Convert.ToDateTime(x.Attributes.FirstOrDefault(y => y.Key == "Data Date").Value))                
                .ToList();
```